include/disruptorplus/spin_wait.hpp

```diff
@@ -108,8 +116,14 @@ namespace disruptorplus
         static void yield_processor()
         {
             // Ideally we want to put this processor into idle mode for a few cycles.
-#if defined(_MSC_VER) && (defined(_M_IX86) || defined(_M_X64))
+#if defined(DISRUPTORPLUS_ARCH_X86)
             _mm_pause();
+#elif defined(DISRUPTORPLUS_ARCH_ARM)
+# if defined(_MSC_VER)
+            __yield();
+# else
+            asm volatile("yield");
```

Comment:
Is `asm` rather than `__asm__` the convention in this library? I'd expect the former to be even more non-standard than the latter.

Reply:
Does not seem to make a huge difference as it's universally supported but since `arm` is descending from gnu mode, I changed it to `__arm__`.